pkg/istio-agent/sds-agent.go

```diff
@@ -344,7 +351,7 @@ func newSecretCache(serverOptions sds.Options) (workloadSecretCache *cache.Secre
 
 			if rootCert, err = ioutil.ReadFile(CitadelCACertPath + "/" + bootstrap.CACertNamespaceConfigMapDataName); err == nil {
 				log.Info("istiod uses self-issued certificate")
-			} else if rootCert, err = ioutil.ReadFile(k8sCAPath); err == nil {
+			} else if rootCert, err = ioutil.ReadFile(k8sCAPath); serverOptions.EnableKubernetesCa && err == nil {
```

Comment:
we are checking this after we read into `rootCert`

Reply:
Added "rootCert = nil" for the else case.